pkg/clients/helm/client.go

```diff
@@ -183,16 +199,31 @@ func (hc *client) pullChart(spec *v1beta1.ChartSpec, creds *RepoCreds, chartDir
 
 	chartRef := spec.URL
 	if spec.URL == "" {
-		chartRef = spec.Name
-
-		pc.RepoURL = spec.Repository
+		if registry.IsOCI(spec.Repository) {
+			chartRef = strings.Join([]string{strings.TrimSuffix(spec.Repository, "/"), spec.Name}, "/")
```

Comment:
Since `spec.Repository` is a url with a protocol using `filepath.Join` will produce
`https:/foo.bar/bar` with a single slash for the protocol. See https://play.golang.com/p/UeoXvtYMAG1
However, I can refactor this to a separate function to make it more clear